document-generator-accounts/src/test/java/uk/gov/companieshouse/document/generator/accounts/AccountsDocumentInfoServiceImplTest.java

```diff
@@ -65,21 +75,30 @@ void testUnsuccessfulGetDocumentInfoNoAccountsResourceInTransaction() throws Ser
         assertNull(accountsDocumentInfoService.getDocumentInfo(createDocumentInfoRequest()));
     }
 
+    @Test
+    @DisplayName("Test DocumentGeneratorInterfaceException thrown when error returned from accounts handler")
+    void testErrorThrownWhenFailedAccountsHandler() throws HandlerException, ServiceException {
+
+        when(transactionService.getTransaction(anyString())).thenReturn(createTransaction());
+        when(accountsHandlerMock.getAbridgedAccountsData(any(Transaction.class),  anyString())).
+                thenThrow(new HandlerException("error"));
+
+        assertThrows(DocumentGeneratorInterfaceException.class, () ->
+                accountsDocumentInfoService.getDocumentInfo(createDocumentInfoRequest()));
+    }
+
     @Test
     @DisplayName("Tests the unsuccessful retrieval of document data due to error in Accounts handler")
     void testUnsuccessfulGetDocumentInfoExceptionFromAccountsHandler()
-            throws HandlerException, ServiceException {
-        when(transactionService.getTransaction(anyString())).thenReturn(createTransaction());
+            throws HandlerException {
         when(accountsHandlerMock.getAbridgedAccountsData(any(Transaction.class),  anyString())).thenThrow(new HandlerException("error"));
 
         assertThrows(HandlerException.class, () -> accountsHandlerMock.getAbridgedAccountsData(transaction, ""));
-
-        assertNull(accountsDocumentInfoService.getDocumentInfo(createDocumentInfoRequest()));
     }
 
     @Test
     @DisplayName("Tests the successful retrieval of document data")
-    void testSuccessfulGetDocumentInfo() throws HandlerException, ServiceException {
+    void testSuccessfulGetDocumentInfo() throws HandlerException, ServiceException, DocumentGeneratorInterfaceException {
```

Comment:
This shouldn't need to catch exceptions from 3 different layers.

Reply:
yeah the handler and service one are needed for;

` when(transactionService.getTransaction(anyString())).thenReturn(createTransaction());
   when(accountsHandlerMock.getAbridgedAccountsData(any(Transaction.class), anyString())).thenReturn(new DocumentInfoResponse());`